src/directLineStreaming.ts

```diff
@@ -0,0 +1,351 @@
+// In order to keep file size down, only import the parts of rxjs that we use
+
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+import { Observable } from 'rxjs/Observable';
+import { Subscriber } from 'rxjs/Subscriber';
+import * as BFSE from 'botframework-streaming';
+
+import {
+  Activity,
+  ConnectionStatus,
+  Conversation,
+  IBotConnection,
+  Media,
+  Message
+} from './directLine';
+
+const DIRECT_LINE_VERSION = 'DirectLine/3.0';
+const MAX_RETRY_COUNT = 3;
+const refreshTokenLifetime = 30 * 60 * 1000;
+const timeout = 20 * 1000;
+const refreshTokenInterval = refreshTokenLifetime / 2;
+
+interface DirectLineStreamingOptions {
```

Comment:
why are all these fields optional? does the library work without them? I look at the implementation and i don't see default values, I see various guards and short-circuiting if these don't exist.

Reply:
I have now marked only the optional fields as optional.